apps/src/code-studio/progress.js

```diff
@@ -234,12 +304,18 @@ function queryUserProgress(store, scriptData, currentLevelId) {
       return;
     }
 
-    // Depend on the fact that even if we have no levelProgress, our progress
-    // data will have other keys
-    const signedInUser = Object.keys(data).length > 0;
+    // If the user is not signed in, retrieve level progress from session storage.
+    // (If the user is signed in, level progress would have been set by the call
+    // to reduxQueryUserProgress above.)
+    if (!data.signedIn) {
```

Comment:
should we also check that user_id === null?

Reply:
Actually, after looking at this some more, I think we probably don't need to check that `user_id === null` here.  This is called from the course overview page and user_id should only be set when the teacher is viewing a student which only applies when the teacher is signed in.  If the user is not signed in, the code should generally be ignoring the user_id parameter even if it is provided (because an anonymous user should not be able to request progress for an arbitrary user).

Reply:
👍